specification/transport/serial/serial.tex

```diff
@@ -0,0 +1,231 @@
+\section{Cyphal/serial (experimental)}\label{sec:transport_serial}
+
+\hyphenation{Cyphal/serial}  % Disable hyphenation.
+
+\subsection{Overview}
+
+This section specifies a concrete transport that operates on top of raw byte-level communication channels,
+such as TCP/IP connections, SSL, UART, RS-232, RS-422, USB CDC ACM,
+and any similar communication links that allow exchange of unstructured byte streams.
+Cyphal/serial may also be used to store Cyphal frames in files.
+\textbf{%
+    As of this version, the Cyphal/serial specification remains experimental.
+    Breaking changes affecting wire compatibility are possible.
+}
+
+As Cyphal/serial is designed to operate over unstructured byte streams,
+it defines a custom framing protocol, custom frame header format, and a custom integrity checking mechanism.
+
+\begin{CyphalSimpleTable}{Cyphal/serial transport capabilities\label{table:transport_serial_capabilities}}{|l X l|}
+    Parameter & Value & References \\
+
+    Maximum node-ID value &
+    65534 (16 bits wide). &
+    \ref{sec:basic} \\
+
+    Transfer-ID mode &
+    Monotonic, 64 bits wide. &
+    \ref{sec:transport_transfer_id} \\
+
+    Number of transfer priority levels &
+    8 (no additional levels). &
+    \ref{sec:transport_transfer_priority} \\
+
+    Largest single-frame transfer payload &
+    Unlimited. &
+    \ref{sec:transport_transfer_payload} \\
+
+    Anonymous transfers &
+    Available. &
+    \ref{sec:transport_route_specifier} \\
+\end{CyphalSimpleTable}
+
+\subsection{Framing}
+
+Cyphal/serial uses the ``Consistent Overhead Byte Stuffing'' (COBS) encapsulation method\footnote{%
+    See ``Consistent Overhead Byte Stuffing'', Stuart Cheshire and Mary Baker.
+    The COBS overhead is 1 byte in every 254 bytes of encapsulated data, which is about 0.4\%.
+} with zero byte as the frame delimiter.
+Due to the nature of COBS, the frame delimiter will not appear in the frame payload.
+A frame delimiter may terminate a frame and/or indicate the start of a new frame.
+The number of frame delimiters between adjacent frames may exceed one.
```

Comment:
What about bit-level framing? Do we want to take advantage of the proposals in the Cheshire and Baker paper for this? If so then the header would start with `0x80, 0x00` (If I'm reading the paper correctly)

Reply:
Bit-level framing is not applicable to byte-level links such as serial port or TCP that this transport is designed for. Unless I misunderstand you.

Reply:
Is that worth discussing in the specification? That Cyphal/Serial expects bit-level framing to be provided by a lower layer? (e.g. you need to use start/stop bits over RS232)

Reply:
The current text states that Cyphal/serial is for byte-level transports, which necessarily implies that lower-level framing is already provided:

>This section specifies a concrete transport that operates on top of raw byte-level communication channels, such as TCP/IP connections, SSL, UART, RS-232, RS-422, USB CDC ACM, and any similar communication links that allow exchange of unstructured byte streams.